src/jshint.js

```diff
@@ -2894,6 +2894,19 @@ var JSHINT = (function() {
 
     functions.push(state.funct);
 
+    // So that the function is available to itself and referencing itself is not
+    // seen as a closure, add the function name to a new scope, but do not
+    // test for unused (unused: false)
+    // it is a new block scope so that params can override it, it can be block scoped
+    // but declarations inside the function don't cause already declared error
+    var internallyAccessibleName = name || classExprBinding;
+    var internallyAccessibleNameType = name ? "function" : "class";
```

Comment:
It looks like we don't need `internallyAccessibleNameType`. Normally I prefer naming values with variables because they serve as documentation, but this one would tend to leak in strange ways--it would be defined as `"class"` even in cases where this was no name binding.